prepare_data/src/test/groovy/org/orbisgis/bdtopo/BDTopoGISLayersTest.groovy

```diff
@@ -41,45 +44,293 @@ class BDTopoGISLayersTest {
-                                    building_bd_topo_use_type: 'BUILDING_BD_TOPO_USE_TYPE' ,
-                                    building_abstract_use_type: 'BUILDING_ABSTRACT_USE_TYPE' ,
-                                    road_bd_topo_type: 'ROAD_BD_TOPO_TYPE', road_abstract_type: 'ROAD_ABSTRACT_TYPE',
-                                    rail_bd_topo_type: 'RAIL_BD_TOPO_TYPE', rail_abstract_type: 'RAIL_ABSTRACT_TYPE',
-                                    veget_bd_topo_type: 'VEGET_BD_TOPO_TYPE', veget_abstract_type: 'VEGET_ABSTRACT_TYPE'
+        assertTrue process.execute([datasource                : h2GISDatabase, tableIrisName: 'IRIS_GE', tableBuildIndifName: 'BATI_INDIFFERENCIE',
+                                    tableBuildIndusName       : 'BATI_INDUSTRIEL', tableBuildRemarqName: 'BATI_REMARQUABLE',
+                                    tableRoadName             : 'ROUTE', tableRailName: 'TRONCON_VOIE_FERREE',
+                                    tableHydroName            : 'SURFACE_EAU', tableVegetName: 'ZONE_VEGETATION',
+                                    distBuffer                : 500, expand: 1000, idZone: '56260',
+                                    building_bd_topo_use_type : 'BUILDING_BD_TOPO_USE_TYPE',
+                                    building_abstract_use_type: 'BUILDING_ABSTRACT_USE_TYPE',
+                                    road_bd_topo_type         : 'ROAD_BD_TOPO_TYPE', road_abstract_type: 'ROAD_ABSTRACT_TYPE',
+                                    rail_bd_topo_type         : 'RAIL_BD_TOPO_TYPE', rail_abstract_type: 'RAIL_ABSTRACT_TYPE',
+                                    veget_bd_topo_type        : 'VEGET_BD_TOPO_TYPE', veget_abstract_type: 'VEGET_ABSTRACT_TYPE'
         ])
         process.getResults().each {
             entry -> assertNotNull(h2GISDatabase.getTable(entry.getValue()))
         }
+
+        // Check if the INPUT_BUILDING table has the correct number of columns and rows
+        def tableName = process.getResults().outputBuildingName
+        def table = h2GISDatabase.getTable(tableName)
+        assertEquals(8, table.columnCount)
+        assertEquals(20568, table.rowCount)
+        // Check if the column types are correct
+        assertEquals('GEOMETRY', table.getColumnsType('THE_GEOM'))
+        assertEquals('VARCHAR', table.getColumnsType('ID_SOURCE'))
+        assertEquals('INTEGER', table.getColumnsType('HEIGHT_WALL'))
+        assertEquals('INTEGER', table.getColumnsType('HEIGHT_ROOF'))
+        assertEquals('INTEGER', table.getColumnsType('NB_LEV'))
+        assertEquals('VARCHAR', table.getColumnsType('TYPE'))
+        assertEquals('VARCHAR', table.getColumnsType('MAIN_USE'))
+        assertEquals('INTEGER', table.getColumnsType('ZINDEX'))
+        // For each rows, check if the fields contains the expected values
+        table.eachRow { row ->
+            assertNotNull(row.THE_GEOM)
+            assertNotEquals('', row.THE_GEOM)
```

Comment:
JUnit assertion are not adapted to the groovy simplification, so if the field is null or '', Junit will only show that the assertion is false instead of true, it won't give details about the error (null or empty value).
In this is (JUnit test), I think it's better to keep the test in two steps.